packages/nodes-base/nodes/Github/__tests__/UrlPatterns.test.ts

```diff
@@ -0,0 +1,241 @@
+				const regex = new RegExp(mode?.extractValue?.regex || '');
+				const url = 'https://github.company.com/my-org';
+				const match = url.match(regex);
+				expect(match?.[2]).toBe('my-org');
+			});
+
+			it('should validate github.com URL', () => {
+				const mode = getOwnerUrlMode();
+				const validation = mode?.validation?.[0] as any;
+				const validationRegex = new RegExp(validation?.properties?.regex ?? '');
+				const url = 'https://github.com/n8n-io';
+				expect(validationRegex.test(url)).toBe(true);
+			});
+
+			it('should validate custom GitHub URL', () => {
+				const mode = getOwnerUrlMode();
+				const validation = mode?.validation?.[0] as any;
+				const validationRegex = new RegExp(validation?.properties?.regex ?? '');
+				const url = 'https://github.company.com/my-org';
+				expect(validationRegex.test(url)).toBe(true);
+			});
+		});
+
+		describe('Repository URL Pattern', () => {
+			const getRepositoryUrlMode = () => {
+				const repoParam = githubTriggerNode.description.properties.find(
+					(prop) => prop.name === 'repository',
+				);
+				return repoParam?.modes?.find((mode) => mode.name === 'url');
+			};
+
+			it('should extract repository from github.com URL', () => {
+				const mode = getRepositoryUrlMode();
+				const regex = new RegExp(mode?.extractValue?.regex || '');
+				const url = 'https://github.com/n8n-io/n8n';
+				const match = url.match(regex);
+				expect(match?.[2]).toBe('n8n');
+			});
+
+			it('should extract repository from custom GitHub URL', () => {
+				const mode = getRepositoryUrlMode();
+				const regex = new RegExp(mode?.extractValue?.regex || '');
+				const url = 'https://github.company.com/my-org/my-repo';
+				const match = url.match(regex);
+				expect(match?.[2]).toBe('my-repo');
+			});
+
+			it('should validate github.com repository URL', () => {
+				const mode = getRepositoryUrlMode();
+				const validation = mode?.validation?.[0] as any;
+				const validationRegex = new RegExp(validation?.properties?.regex ?? '');
+				const url = 'https://github.com/n8n-io/n8n';
+				expect(validationRegex.test(url)).toBe(true);
+			});
+
+			it('should validate custom GitHub repository URL', () => {
+				const mode = getRepositoryUrlMode();
+				const validation = mode?.validation?.[0] as any;
+				const validationRegex = new RegExp(validation?.properties?.regex ?? '');
+				const url = 'https://github.company.com/my-org/my-repo';
+				expect(validationRegex.test(url)).toBe(true);
+			});
+		});
+	});
+
+	describe('URL Pattern Edge Cases', () => {
+		it('should handle URLs with subdomains', () => {
+			const ownerPattern = /https:\/\/([^/]+)\/([-_0-9a-zA-Z]+)/;
+			const repoPattern = /https:\/\/([^/]+)\/(?:[-_0-9a-zA-Z]+)\/([-_.0-9a-zA-Z]+)/;
+
+			// Test complex custom URLs
+			expect('https://git.internal.company.com/dev-team'.match(ownerPattern)?.[2]).toBe('dev-team');
+			expect('https://github.acme.corp/engineering/backend-api'.match(repoPattern)?.[2]).toBe(
```

Comment:
Same as with previous tests
```suggestion
			expect('https://git.internal.company.com/dev-team'.match(ownerPattern)?.[1]).toBe('dev-team');
			expect('https://github.acme.corp/engineering/backend-api'.match(repoPattern)?.[1]).toBe(
```